File: api/schemas/staging.py
"""Pydantic schemas for staging endpoints."""

from __future__ import annotations

from pydantic import BaseModel


class StagingItem(BaseModel):
    """Single pending item in the staging queue."""

    sha256: str
    filename: str
    size_bytes: int
    first_seen_at: str
    updated_at: str
    account: str | None = None
    onedrive_id: str | None = None


class StagingPage(BaseModel):
    """Paginated list of pending items."""

    items: list[StagingItem]
    cursor: str | None = None
    has_more: bool
    total: int

File: webui/src/lib/api/staging.ts
import { apiFetch } from './client';

export interface StagingItem {
  item_id: string;
  filename: string;
  sha256: string;
  first_seen_at: string;
  account: string;
  status: string;
}

export interface StagingPage {
  items: StagingItem[];
  total: number;
  cursor: string | null;
}

export function getStagingPage(cursor?: string | null, limit = 20): Promise<StagingPage> {
  const params = new URLSearchParams();
  params.set('limit', String(limit));
  if (cursor) {
    params.set('after', cursor);
  }
  return apiFetch<StagingPage>(`/api/v1/staging?${params.toString()}`);
}

export function getItem(id: string): Promise<StagingItem> {
  return apiFetch<StagingItem>(`/api/v1/items/${encodeURIComponent(id)}`);
}
